tests/functional/test_install_wheel.py

```diff
@@ -388,7 +406,12 @@ def test_wheel_record_lines_have_updated_hash_for_scripts(
 
     script_path = script.bin_path / "dostuff"
     script_contents = script_path.read_bytes()
-    assert not script_contents.startswith(b"#!python\n")
+    expected_prefix = (
+        b"#!/bin/sh\n'''exec'"
```

Comment:
Strong -1 on having the test rely on the precise mechanism used in the generated shebang..

Reply:
how should I do it? without knowledge of the mechanism, checks like `exe_path in script_contents` aren’t reliable either, because maybe the space is escaped with `\` or so.

Reply:
I don't know. It's a hard problem. Distlib has a bunch of tests around shebangs, so if we were using documented and supported distlib APIs, I'd say we could rely on them (and the existing check, that just confirms we did rewrite, would be enough). But because we're using the internal `_get_shebang` function, we can't do that, as we have no way to ensure that we're using the function in a way that is covered by the distlib tests.

The fact that tests are failing on Windows demonstrates that this test isn't sufficient - the shebang you're checking for isn't what gets used on Windows (because Windows shells don't support it).